### PR TITLE
Calculator edit form

### PR DESCRIPTION
**Pull request by OAGr**

### CODE REVIEW

**Comment by OAGr** [comment]
Things to fix:
- User should not be able to create calculators for models they cannot edit.
- Metrics with empty names shouldn't show in calculator edit sidebar
- Intermediate metrics should show as 'outputs'

**Comment by mmcdermott** [comment]
Bug PRs:
* Elev hidden: #622

File: src/components/calculators/new/new-calculator-selector.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// TODO(matthew): DRY this code up.
+function _sameId(idA, idB){
+  return idA.toString() === idB.toString()
+}
+
+const INTERMEDIATE = 'INTERMEDIATE'
+const OUTPUT = 'OUTPUT'
+const INPUT = 'INPUT'
+const NOEDGE = 'NOEDGE'
```
Comment (OAGr) [inline src/components/calculators/new/new-calculator-selector.js]: This should be dried up, as in helper

File: src/components/calculators/show/CalculatorShow.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ import * as Calculator from 'gEngine/calculator'
 
 import {Guesstimator} from 'lib/guesstimator/index'
 
-import './style.css'
+import '../shared/style.css'
```
Comment (OAGr) [inline src/components/calculators/show/CalculatorShow.js]: not sure if this is needed, as if it is shared it should already be imported

File: src/components/distributions/editor/TextForm/TextInput.js [hunk 3]
```diff
@@ -4,31 +4,70 @@ import $ from 'jquery'
...
     },
+    extraDecorators: [],
+    decoratorsUpToDate: false,
   }
 
   static propTypes = {
     value: PropTypes.string,
   }
 
+  decoratorList() {
```
Comment (OAGr) [inline src/components/distributions/editor/TextForm/TextInput.js]: Are the decorator changes supposed to be part of this PR?

Comment (mmcdermott) [inline src/components/distributions/editor/TextForm/TextInput.js]: Not particularly, but as they're already merged to master it can't hurt, right?

Comment (mmcdermott) [inline src/components/distributions/editor/TextForm/TextInput.js]: Not particularly, but as they're already merged to master, it doesn't seem like a bad thing

**Pull request merged**